src/core/public/application/application_service.tsx

```diff
@@ -131,6 +174,14 @@ export class ApplicationService {
       },
 
       navigateToApp: (appId, { path, state }: { path?: string; state?: any } = {}) => {
+        if (this.apps.get(appId) === undefined) {
+          throw new Error(`Trying to navigate to an unknown application: ${appId}`);
+        }
+        const app = availableApps$.value.get(appId);
+        if (app === undefined || app.status !== AppStatus.accessible) {
+          throw new Error(`Trying to navigate to an inaccessible application: ${appId}`);
+        }
```

Comment:
What is our strategy regarding error throwing ? Should I create a custom error type every time, even when not enhancing `Error` with custom props?

Reply:
That's a good question. `navigateToApp` is a part of public API and can be called by any plugin. If we throw an error here, it becomes a plugin responsibility to handle the error. I think the core should handle core errors.

Reply:
So instead of throwing, we should probably redirect to a 404 page or something similar ?

Reply:
or probably call FatalError https://github.com/pgayvallet/kibana/blob/09cd6d82af8dd192036d96cabe803c4ce7615a4f/src/core/public/fatal_errors/fatal_errors_service.tsx#L49
Although it doesn't scale well. We can forget to wrap the public method, for example. Probably it worth a separate discussion.

Reply:
One option in this case is to redirect to the route (basically, remove this check) and rely on the existing "Application Not Found" UI to show. That would be a bit more graceful than breaking the calling code, but may obscure a bug. Though this case also exposes the need for plugins to know which applications are available in order to show/hide links to other apps.

Reply:
> Though this case also exposes the need for plugins to know which applications are available in order to show/hide links to other apps.

Have they got other links besides navLinks?
The platform could provide a list of available apps. Then plugins can use this info to enable/disable links. Should they hide links or make them disabled (for the sake of UI consistency)?
For now `Application Not Found` is ok. And it's inlined with behavior when a user lands on URL for an unknown or disabled app.

Reply:
removed the error logic, user will now lands on the 404 page when redirecting to a non-existing app